### DISCUSSION TITLE
Docker hosting migration (1.0.58+)

### DISCUSSION THREAD

**Discussion by Rory-Powell** [Budibase/budibase#5169] [General]
Hello all, this discussion outlines the steps required for migrating to the latest docker-compose.yaml from versions 1.0.58 onwards. 

## What changed
In this version envoy proxy has been replaced with nginx

For more details see the Issue / PR:
- https://github.com/Budibase/budibase/issues/2274
- https://github.com/Budibase/budibase/pull/4268

### TLDR
- `hosting.properties` has been replaced with `.env`
- `docker-compose.yaml` has been updated to use the `budibase/proxy` which is nginx + custom configuration
<img width="400" alt="Screenshot 2022-03-30 at 08 15 00" src="https://user-images.githubusercontent.com/8755148/160773452-3ff05873-7ea6-4fde-842c-35764a27152b.png">

## Migration steps

- Go to your install directory where `docker-compose.yaml` is located
- Rename `hosting.properties` to `.env`

The next step can be completed manually or using the budibase cli

### Manually
Replace the contents of `docker-compose.yaml` with the latest
https://raw.githubusercontent.com/Budibase/budibase/master/hosting/docker-compose.yaml

### CLI
First, make sure your CLI is up to date. Download the latest here:
https://docs.budibase.com/docs/budibase-cli-setup

Or if you are using the CLI as a node package, re-run the install script:
```
npm i -g @budibase/cli
```

Now you can run the update command. 
```
budi hosting --update
```
When asked `? Do you wish to update you docker-compose.yaml? (Y/n)`, type `Y`. 

## When do I need to migrate

Versions `1.0.58` through to `1.0.97` should function as normal while still using envoy proxy.
In version `1.0.98` onwards the nginx migration is **required.**

**Comment by Rory-Powell** [comment]
See the above steps for migrating to the latest docker-compose.yaml

**Reply by the-real-jeremy-coleman** [reply]
Am I understanding it correctly that going forward budibase on docker for self-hosting will require nginx? If that's the case, how difficult would it be for me to strip that requirement and run without nginx? I just recently discovered budibase and was hoping to start using it to increase my teams' ability to respond to changing data needs within our organization. We have a hard requirement that prohibits the use of nginx due to its Russian origin. I'm not here to argue the merits or politics of that requirement, it's non-negotiable and above my level. I just need to know if this is a breaking change which means I would be stuck at v 1.0.57 with no way to move forward.

**Reply by shogunpurple** [reply]
@mammlouk thats true - we will be using NGINX going forward and it’s unlikely that will change. 

To get around it, you could translate the NGINX config to another reverse proxy, but that would require you to maintain it yourself. Caddy has a tool that helps to automate this:
https://github.com/caddyserver/nginx-adapter

But not sure how reliable it is. I apologise that this has affected your installation - it does seem like a fairly stringent requirement given that NGINX is owned by an American company, and the Russian engineer who created it originally no longer works there as of this year. 

Hope this helps.

**Comment by KernOil** [comment]
Not sure what I am doing wrong. I'm hosing on docker and I have updated the docker-compose.yaml and I have updated the CLI and run through all of the instructions on both of these pages. I am still only able to view the last published app not any of my other apps.

**Reply by Rory-Powell** [reply]
Hi @KernOil. Some ideas:
- Do you have any modifications to your docker-compose file or has it been copied exactly?
- Can you try the manual step of copying over the compose instead of using the CLI?
- What error logs are you seeing when accessing your app?

**Reply by KernOil** [reply]
HI, I have tried updating with CLI and I have also tried manual copy and paste of the docker-compose file. 
my log file is not showing any errors. when I open any published app it will only show me the most recent published app. basically across my 4 published apps the same app opens when clicking on any of the 4 app icons.

**Reply by KernOil** [reply]
https://user-images.githubusercontent.com/80058238/161791433-b4eeeb82-3cb0-4653-8421-6ba539cae3e0.mp4

**Reply by KernOil** [reply]
I fixed it by downloading all my apps deleting everything then rebuilding BB from scratch. working great now

**Comment by reichert** [comment]
The same problem occurs if you have multi tenant activated.
I deactivated it and the http 500 - no database found just was solved.

![image](https://user-images.githubusercontent.com/1287032/171210110-01dbd68f-071d-4f38-bbb9-f2f324827813.png)

**Reply by Rory-Powell** [reply]
@reichert there are some steps required in order to make tenancy work involving DNS names, in future there will be guides on how to do this but right now multi tenancy isn't supported in self hosted installations.
